nixGL を現在の nixpkgs でビルドできるようにする。
nixpkgs 側で nvidia_x11 が kernel 引数を受け取らなくなり、Vulkan ICD の
ファイル名も nvidia_icd.x86_64.json から nvidia_icd.json へ変わったため、
その2点を修正する。上流の未マージ PR と同じ内容。
https://github.com/nix-community/nixGL/pull/223

--- a/nixGL.nix
+++ b/nixGL.nix
@@ -87,7 +87,6 @@
 
       nvidiaLibsOnly = nvidiaDrivers.override {
         libsOnly = true;
-        kernel = null;
       };
 
       nixGLNvidiaBumblebee = writeExecutable {
@@ -131,10 +130,12 @@
 
               ${
                 lib.optionalString (api == "Vulkan")
-                ''export VK_ICD_FILENAMES=${nvidiaLibsOnly}/share/vulkan/icd.d/nvidia_icd.x86_64.json${
+                ''NVIDIA_ICD=(${nvidiaLibsOnly}/share/vulkan/icd.d/*.json${
                   lib.optionalString enable32bits
-                  ":${nvidiaLibsOnly.lib32}/share/vulkan/icd.d/nvidia_icd.i686.json"
-                }"''${VK_ICD_FILENAMES:+:$VK_ICD_FILENAMES}"''
+                  " ${nvidiaLibsOnly.lib32}/share/vulkan/icd.d/*.json"
+                })
+                NVIDIA_ICD_FILENAMES=$(IFS=:; echo "''${NVIDIA_ICD[*]}")
+                export VK_ICD_FILENAMES="''${NVIDIA_ICD_FILENAMES}''${VK_ICD_FILENAMES:+:$VK_ICD_FILENAMES}"''
               }
               export LD_LIBRARY_PATH=${
                 lib.makeLibraryPath ([ libglvnd nvidiaLibsOnly ]
